projects/electron-ngrx/src/lib/selector-manager.ts

```diff
@@ -0,0 +1,22 @@
+import {h64} from 'xxhashjs';
+
+type fooType = (...args: any) => any;
+const hashFunction = (foo: fooType, identifier: string): string => {
+  return h64(foo.name + foo.toString() + identifier, 0xABCD).toString(16);
+};
+let selectorStorage: Record<string, fooType> = {};
+export const registerSelector = (selector, identifier = '') => {
+  const hash = hashFunction(selector, identifier);
+  if (selectorStorage.hasOwnProperty(hash)) {
+    throw new Error(`duplicate selector.\
```

Comment:
If you want your library to be professional. You shouldn't just add capital letters out of nowhere.
Use this:  "Duplicate selector identified. The selector: '${selector.name}' already exists. If you didn't register this selector a naming collision has occurred. To fix this error call the registerSelector with the second argunent as an identifier to the selector".